test/commands.js

```diff
@@ -564,6 +564,58 @@ let runFundCrowdsaleBelowSoftCap = async (command, state) => {
   return state;
 }
 
+let runFundCrowdsaleOverSoftCap = async (command, state) => {
+  if (!state.crowdsaleFinalized) {
+    // unpause the crowdsale if needed
+    if (state.crowdsalePaused) {
+      state = await runPauseCrowdsaleCommand({pause: false, fromAccount: state.owner}, state);
+    }
+
+    // set weiPerUSDinTGE rate if needed
+    if (state.weiPerUSDinTGE == 0) {
+      state = await runSetWeiPerUSDinTGECommand({wei: 10000, fromAccount: state.owner}, state);
+    }
+
+    let softCap = await state.crowdsaleContract.maxFoundationCapUSD.call(),
+      currentUSDFunding = state.weiRaised.div(state.weiPerUSDinTGE);
+
+    if (softCap > currentUSDFunding) {
+      // wait for crowdsale startTimestamp
+      if (latestTime() < state.crowdsaleData.startTimestamp) {
+        await increaseTimeTestRPCTo(state.crowdsaleData.startTimestamp);
+      }
+
+      // buy enough tokens to exactly reach the minCap (which is less than softCap)
+      let wei = softCap.minus(currentUSDFunding).mul(state.weiPerUSDinTGE).plus(command.softCapExcessWei),
+        eth = web3.fromWei(wei, 'ether'),
+        buyTokensCommand = {account: command.account, eth: eth, beneficiary: command.account};
+
+      state = await runBuyTokensCommand(buyTokensCommand, state);
+
+      wei.should.be.bignumber.equal(state.weiRaised);
+    }
+
+    if (command.finalize) {
+      // wait for crowdsale end2Timestamp
+      if (latestTime() < state.crowdsaleData.end2Timestamp) {
+        await increaseTimeTestRPCTo(state.crowdsaleData.end2Timestamp + 1);
```

Comment:
its necessary the +1 ?

Reply:
yes, because we want it to be after the `end2Timestamp`, only after it the crowdsale can be finalized

Reply:
:+1: